### ISSUE TITLE
ratings graph edges

### ISSUE THREAD

**Issue by A1Liu** [a2liu/DSFP#14]
Each node in the ratings graph should have a collection of edges going from it to other nodes that have rated it, instead of the other way around. The weight of the edges will still be a rating, but it'll be a rating that's received instead of given. Lists of ratings for the user can be generated through SQL queries instead of through the network; the network is built for searching for friends and for updating the ratings table.

**Comment by A1Liu** [comment]
So like the edgelist for a node will have edges from this edge to other edges, but the "destination" field will be the source of the rating, and the weight will be the rating that the other node has given (or that this node has been given). This way, updating the ratings network becomes a simpler special case of breadth-first traversal.

**Comment by A1Liu** [comment]
Also the edges might have to contain multiple values to signify how much of a weight nodes had before updates.

**Comment by A1Liu** [comment]
TLDR: Nodes need to contain references to the nodes that have rated them so that calculations can be done.
